modules/graceful/server.go

```diff
@@ -128,6 +128,8 @@ func (srv *Server) ListenAndServeTLS(certFile, keyFile string, serve ServeFuncti
 func (srv *Server) ListenAndServeTLSConfig(tlsConfig *tls.Config, serve ServeFunction) error {
 	go srv.awaitShutdown()
 
+	tlsConfig.MinVersion = tls.VersionTLS12
```

Comment:
This should be set before when config is creating around here:
https://github.com/go-gitea/gitea/blob/c512bfd09c97b0b8ee8e5f916e7a2ad0750301e1/modules/graceful/server.go#L123

Also it would be better if this could be configured in app.ini so that it could be disable in case when needed to use with older browsers or integrations that do not support TLS 1.2

Reply:
I intentionally set the restriction here to make sure it's enforced across all TLS connections ever made by Gitea. Given the security issues present in TLS 1.0 and 1.1, it doesn't make much sense to run TLS with vulnerable versions instead of not running it.

Reply:
Even vulnerable security is better than no security and I agree that default should be secure by default but it would be still better to allow for users to change it in case they need it for backward compatibility

Reply:
OK, what is the best way to expose this via `app.ini`?

Reply:
I think same way as password hashing algorithm

Reply:
I'll work on it and follow up with an update, thank you.